src/utils/tree_node.C

```diff
@@ -485,40 +493,33 @@ const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_i
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element_in_children(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem* TreeNode<N>::find_element_in_children
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   libmesh_assert (!this->active());
 
-  unsigned int excluded_child = libMesh::invalid_uint;
+  std::vector<bool> searched_child(children.size(), false);
```

Comment:
Yeah, but I'm not even talking about read/write race conditions.  There can be a data race in *read-only* threaded code that uses vector<bool>, as I understand it, and we have in fact encountered it before in MOOSE.  

Not to mention that there is apparently [a general consensus among the C++ Standard Committee and the Library Working Group that vector<bool> should be deprecated](http://en.wikipedia.org/wiki/Sequence_container_(C%2B%2B)#Specialization_for_bool)